drf_spectacular/contrib/rest_framework_dataclasses.py

```diff
@@ -0,0 +1,26 @@
+from drf_spectacular.extensions import OpenApiSerializerExtension
+from drf_spectacular.plumbing import get_doc
+from drf_spectacular.utils import Direction
+
+
+class OpenApiDataclassSerializerExtensions(OpenApiSerializerExtension):
+    target_class = "rest_framework_dataclasses.serializers.DataclassSerializer"
+    match_subclasses = True
+
+    def get_name(self):
+        """Use the dataclass name in the schema, instead of the serializer prefix (which can be just Dataclass)."""
+        return self.target.dataclass_definition.dataclass_type.__name__
+
+    def strip_library_doc(self, schema):
+        """Strip the DataclassSerializer library documentation from the schema."""
+        if not hasattr(self, '_library_doc'):
```

Comment:
good thinking, but there is no point in caching this in the instance, as a new instance is created on every new match.

Reply:
Ah, well, let's just extract it everytime then.